samtranslator/model/eventsources/push.py

```diff
@@ -429,11 +429,12 @@ def _get_permission(self, resources_to_link, stage, suffix):
             raise RuntimeError("Could not add permission to lambda function.")
 
         path = self.Path.replace('{proxy+}', '*')
+        method = '*' if self.Method == 'any' else self.Method.upper()
```

Comment:
This may work but can be easily broken. I am not sure if self.Method is normalized (all upper or lower) but we should just force one of these here. It will also help in readability.

Reply:
It's not obvious, but this is already being forced to lower here https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/eventsources/push.py#L399

Reply:
Let's make it obvious here then :)

Reply:
So I should make it like `self.Method.lower() == 'any'`?

Reply:
`.lower()` or `.upper()`. Which one doesn't explicitly matter, just as long as it matches what you compare it to (`any` or `ANY`).